com.unity.netcode.gameobjects/Components/NetworkAnimator.cs

```diff
@@ -797,31 +789,32 @@ internal void CheckForAnimatorChanges()
                     continue;
                 }
 
-                var animationState = new AnimationState
-                {
-                    IsDirty = true,
-                    Transition = false, // Only used during synchronization
-                    StateHash = stateHash,
-                    NormalizedTime = normalizedTime,
-                    Layer = layer,
-                    Weight = m_LayerWeights[layer]
-                };
+                // We use the IsDirtyCount here because we are only updating the AnimationStates that
+                // need to be updated
+                var animationState = m_AnimationMessage.AnimationStates[m_AnimationMessage.IsDirtyCount];
```

Comment:
Echoes to my above comment. How is it that the count is always the one we want to index? It gives the impression that `AnimationStates` is actually a list of state to synchronize and, here, we mean to access the last? The the naming is misleading. `AnimationStates` conveys it being indexed by something logical around the animation itself.

Maybe I'm just unfamiliar, of maybe it needs a comment block at the top to explain what is indexed by what.